### ISSUE TITLE
Plato should support being built with a mupdf version with different static arrayed fallback fonts prefixes

### ISSUE THREAD

**Issue by Aszarsha** [baskerville/plato#99]
On Arch Linux, inside mupdf, the inlined fallback fonts which `src/font/mod.rs` depends on have a different prefix, leading to plenty of undefined references.
- thirdparty built lib: `_binary_DroidSansFallback_ttf` and `_binary_DroidSansFallback_ttf_size`
- Arch Linux mupdf 1.16.1 lib: `_binary_resources_fonts_droid_DroidSansFallback_ttf_end`, `_binary_resources_fonts_droid_DroidSansFallback_ttf_size` and `_binary_resources_fonts_droid_DroidSansFallback_ttf_start`

Maybe updating thirdparty to 1.16.1 will solve the issue.

Furthermore, maybe Plato should be able to only rely on external fonts such as the ones provided in the `/mnt/onboard/fonts` or /mnt/onboard/.adds/Plato/fonts` directories as well as the ones bundled in the documents.

**Comment by baskerville** [comment]
> On Arch Linux, inside mupdf, the inlined fallback fonts which `src/font/mod.rs` depends on have a different prefix, leading to plenty of undefined references.
> 
> * thirdparty built lib: `_binary_DroidSansFallback_ttf` and `_binary_DroidSansFallback_ttf_size`
> * Arch Linux mupdf 1.16.1 lib: `_binary_resources_fonts_droid_DroidSansFallback_ttf_end`, `_binary_resources_fonts_droid_DroidSansFallback_ttf_size` and `_binary_resources_fonts_droid_DroidSansFallback_ttf_start`
> 
> Maybe updating thirdparty to 1.16.1 will solve the issue.

It won't: the *\*_start*–*\*_end*–*\*_size* symbols are used on Linux, the other platforms use the other naming scheme. This is an upstream issue.

I'll see if I can work around the problem using `#[cfg(target_os = …`

> Furthermore, maybe Plato should be able to only rely on external fonts such as the ones provided in the `/mnt/onboard/fonts` or /mnt/onboard/.adds/Plato/fonts` directories as well as the ones bundled in the documents.

The font fallback mechanism is based on the fonts embedded in libmupdf.

**Comment by baskerville** [comment]
Could you try the [libmupdf-objcopy-symbols](https://github.com/baskerville/plato/tree/libmupdf-objcopy-symbols) branch?

**Comment by Aszarsha** [comment]
> Could you try the [libmupdf-objcopy-symbols](https://github.com/baskerville/plato/tree/libmupdf-objcopy-symbols) branch?

Sure. I get:
```
error[E0308]: mismatched types
   --> src/font/mod.rs:562:54
    |
562 | unsafe fn font_data_from_script(script: HbScript) -> &'static [libc::c_uchar] {
    |           ---------------------                      ^^^^^^^^^^^^^^^^^^^^^^^^ expected `&[u8]`, found `()`
    |           |
    |           implicitly returns `()` as its body has no tail or `return` expression
```

See my comments on the commit, target with _r_. ;-)

I'm getting linking error, I'll merge on top of my [emulator_building](/Aszarsha/plato/tree/emulator_building) branch and see how it goes.

**Comment by Aszarsha** [comment]
Fixed by #101 which merges [libmupdf-objcopy-symbols](https://github.com/baskerville/plato/tree/libmupdf-objcopy-symbols).
